CDCgov/prime-reportstream
# HL7 OBX Mapping for OML messages

This PR adds mapping to HL7 `OBX` fields for the `OML` message type. It also extracts the common field definitions for `ORU` and `OML` into a common file to be reused (`metadata/hl7_mapping/common/observation-result.yml`)

Test Steps:
1. Run `./gradlew primeCLI --args='fhirdata --output-format HL7 --input-file "./src/testIntegration/resources/datatests/FHIR_to_HL7/sample_OML_20230831-0001.fhir"  -s "./metadata/hl7_mapping/receivers/Flexion/TILabOrder.yml"'`
2. Check that the output HL7 message has the `OBX` field. Subfields to look for are `OBX-1`, `OBX-2`, `OBX-3`, `OBX-4`, `OBX-5`, `OBX-6`, `OBX-11` and `OBX-14`

## Changes
- Extracted common field definitions in `ORU` and `OML` to `metadata/hl7_mapping/common/observation-result.yml`
- Added condition to `OBX-2` and `OBX-5` to support `NM`/`Quantity` data type
- Removed duplicated definition for `observation-identifier-code` field

## Checklist

### Testing

## Linked Issues
[Create Outbound OML OBX Segment](https://github.com/CDCgov/trusted-intermediary/issues/495)
